### PR TITLE
Add grafanalib helpers

### PR DESCRIPTION
**Pull request by francoisferrand**
Issue: ZENKO-3900

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install -e .[dev]
+      - name: Format with black
+        run: black --diff --check $(git ls-files '*.py')
+      - name: Lint with pylint
+        run: pylint --disable=all --enable=unused-import $(git ls-files '*.py')
+      - name: Check types with mypy
+        run: mypy $(git ls-files '*.py')
```
Comment (lavigne958) [inline .github/workflows/lint.yaml]: missing trailing end new line

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          python-version: 3.x
+      - name: Install dependencies
+        run: |
+          pip install --upgrade pip
+          pip install -e .[dev]
+      - name: Format with black
+        run: black --diff --check $(git ls-files '*.py')
+      - name: Lint with pylint
+        run: pylint --disable=all --enable=unused-import $(git ls-files '*.py')
```
Comment (jbertran) [inline .github/workflows/lint.yaml]: Pylint only checks unused imports here :upside_down_face:

**Review by lavigne958 — APPROVED** [review]

File: scalgrafanalib/core.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    show: bool = attr.ib(default=True, validator=attr.validators.instance_of(bool))
+    showHistogram: bool = attr.ib(  # pylint: disable=invalid-name
+        default=True, validator=attr.validators.instance_of(bool)
+    )
+
+    def to_json_data(self) -> Json:
+        return {"show": self.show, "showHistogram": self.showHistogram}
+
+
+@attr.s
```
Comment (jbertran) [inline scalgrafanalib/core.py]: Do we need the decorator when the class has no attributes, just the method redefinition?

File: scalgrafanalib/core.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    def to_json_data(self) -> Json:
+        json = super().to_json_data()
+        if self.decimals:
+            json["options"]["decimals"] = self.decimals
+        return json
+
+
+def _simplify(json: Json) -> Json:
+    return dict(filter(lambda item: item[1] is not None, json.items()))
```
Comment (jbertran) [inline scalgrafanalib/core.py]: pythonic way: `{ key: value for key, value in json.items() if value is not None }

File: scalgrafanalib/layout.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        return 1
+    if panel.gridPos is None:
+        return 0
+    return panel.gridPos.h
+
+
+def get_width(panel: Panel) -> int:
+    """Return the width of the panel"""
+    if isinstance(panel, RowPanel):
+        return 24
```
Comment (jbertran) [inline scalgrafanalib/layout.py]: As much as `1` and `0` sound like sane defaults, we don't know where this `24` comes from. Might be worth setting all 3 as constants (i.e. defined at top-level, all caps)

File: scalgrafanalib/layout.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    if panel.gridPos is None:
+        return 0
+    return panel.gridPos.w
+
+
+def row(panels: PanelList, height: int, width: int = 0) -> PanelRow:
+    """Resize panels so they are evenly spaced."""
+    if width == 0:
+        alloted_widths = list(filter(None, [get_width(panel) for panel in panels]))
+        width = floordiv(
```
Comment (jbertran) [inline scalgrafanalib/layout.py]: `floordiv` is the same as `//`, no?

**Review by jbertran — COMMENTED** [review]
Just a few comments. I'm still not a fan of `attrs`, but I suspect it's just because I haven't completely understood the amount of things it does.

We should create a ticket for adding tests - I'm clicking on faith for the height/width computations :sweat_smile:

**Review by jbertran — APPROVED** [review]

**Review by lavigne958 — APPROVED** [review]

**Pull request merged**